src/main.rs

```diff
@@ -53,6 +54,37 @@ fn show_settings(
     );
 }
 
+#[derive(Serialize)]
+struct Explanation<'a> {
+    code: &'a str,
+    category: &'a str,
+    summary: &'a str,
+}
+
+fn explain(code: &CheckCode, format: SerializationFormat) -> Result<()> {
+    match format {
+        SerializationFormat::Text => {
+            println!(
+                "{} ({}): {}",
+                code.as_ref(),
+                code.category().title(),
+                code.kind().summary()
+            );
+        }
+        SerializationFormat::Json => {
+            println!(
+                "{}",
+                serde_json::to_string_pretty(&Explanation {
+                    code: code.as_ref(),
+                    category: code.category().title(),
+                    summary: &code.kind().summary(),
+                })?
+            );
+        }
```

Comment:
We might not need json. For https://github.com/charliermarsh/vscode-ruff/pull/26, just showing the text representation is enough for now.